### PR TITLE
Mk single pion

### PR DESCRIPTION
**Pull request by mroda88**
This is the branch candidate for the Minoo's model based on Igor's work and with some less invasive approach for the rest of the code.

### CODE REVIEW

File: config/EventGenerator.xml [hunk 1]
```diff
@@ -123,18 +123,6 @@ XSecModel                   alg     Yes  Cross section model used at the thread
   </param_set>
 
 
- <!--
```
Comment (mroda88) [inline config/EventGenerator.xml:126]: This should not be deleted

File: config/EventGenerator.xml [hunk 2]
```diff
@@ -174,25 +162,8 @@ XSecModel                   alg     Yes  Cross section model used at the thread
 
 
    <!--
-   CCQE generator for SuSA-1p1h which is loosely based on the MEC-CC generator-->
-   <!--
```
Comment (mroda88) [inline config/EventGenerator.xml:178]: This should not be deleted

File: config/EventGenerator.xml [hunk 3]
```diff
@@ -670,73 +682,4 @@ XSecModel                   alg     Yes  Cross section model used at the thread
...
-     <param type="alg"    name="Module-2">   genie::HEDISKinematicsGenerator/Default           </param>
-     <param type="alg"    name="Module-3">   genie::HEDISGenerator/Default                        </param>
-     <param type="alg"    name="Module-4">   genie::UnstableParticleDecayer/BeforeHadronTransport </param>
-     <param type="alg"    name="ILstGen">    genie::HEDISInteractionListGenerator/CC-Default      </param>
-  </param_set>
-
-  <param_set name="HEDIS-NC"> 
-     <param type="string" name="VldContext"> </param>
-     <param type="int"    name="NModules">   5                                                    </param>
-     <param type="alg"    name="Module-0">   genie::InitialStateAppender/Default                  </param>
```
Comment (mroda88) [inline config/EventGenerator.xml:689]: This should not be deleted

File: config/Messenger.xml [hunk 4]
```diff
@@ -140,7 +146,7 @@
   <priority msgstream="LwlynSmith">            NOTICE </priority>
   <priority msgstream="LHAPDF6">               NOTICE </priority>
   <priority msgstream="Math">                  NOTICE </priority>
-  <priority msgstream="MEC">                   WARN  </priority>
```
Comment (mroda88) [inline config/Messenger.xml:143]: This should not change

File: config/Messenger.xml [hunk 5]
```diff
@@ -201,15 +211,6 @@
   <priority msgstream="XSecAlgMap">            NOTICE </priority>
   <priority msgstream="XSecSplLst">            NOTICE </priority>
   <priority msgstream="ZExpAxialFormFactorModel"> NOTICE </priority>
-  <priority msgstream="GLRESXSec">             WARN    </priority>
```
Comment (mroda88) [inline config/Messenger.xml:204]: these should not be removed

File: config/Messenger.xml [hunk 6]
```diff
@@ -100,6 +105,7 @@
   <priority msgstream="GuiUtils">              INFO   </priority>
   <priority msgstream="GRV98LO">               WARN   </priority>
   <priority msgstream="GXSecFunc">             WARN   </priority>
+  <priority msgstream="GLRES">                 WARN   </priority>
```
Comment (mroda88) [inline config/Messenger.xml:108]: This should not be added

File: config/EventGenerator.xml [hunk 7]
```diff
@@ -476,6 +445,15 @@ XSecModel                   alg     Yes  Cross section model used at the thread
      <param type="alg"    name="ILstGen">    genie::AMNuGammaInteractionListGenerator/Default    </param>
   </param_set>
 
+  <param_set name="GLRES">
+     <param type="string" name="VldContext"> </param>
+     <param type="int"    name="NModules">   3                                                   </param>
```
Comment (mroda88) [inline config/EventGenerator.xml:450]: This should not be added

File: src/Framework/ParticleData/BaryonResUtils.h [hunk 8]
```diff
@@ -5,10 +5,20 @@
 
 \brief     Baryon Resonance utilities.
 
-\author    Costas Andreopoulos <constantinos.andreopoulos \at cern.ch>
-           University of Liverpool & STFC Rutherford Appleton Laboratory
 
-\created   November 25, 2004
+  \authors    Costas Andreopoulos <costas.andreopoulos \at stfc.ac.uk>
```
Comment (mroda88) [inline src/Framework/ParticleData/BaryonResUtils.h:9]: Restore old author

**Comment by mroda88** [comment]
We need to check the following
 - [ ] `gspl2root` is compatible with the new splines

File: src/Physics/Resonance/XSection/MKSPPXSecWithCache.cxx [hunk 9]
```diff
@@ -0,0 +1,283 @@
...
+  Cache * cache = Cache::Instance();
+
+  assert(fSingleResXSecModel);
+  
+  SppChannel_t spp_channel = SppChannel::FromInteraction(in);
+
+  // Compute the number of spline knots - use at least 10 knots per decade
+  // && at least 40 knots in the full energy range
+  const double Emin       = 0.01;
+  const int    nknots_min = (int) (10*(TMath::Log(fEMax)-TMath::Log(Emin)));
```
Comment (mroda88) [inline src/Physics/Resonance/XSection/MKSPPXSecWithCache.cxx:91]: The usage of this fEMax is a **major** issue. We need to make the system more consistent with the configuration.

File: src/Physics/Resonance/XSection/MKSPPXSecWithCache.cxx [hunk 10]
```diff
@@ -0,0 +1,283 @@
...
+
+  Cache * cache = Cache::Instance();
+
+  assert(fSingleResXSecModel);
+  
+  SppChannel_t spp_channel = SppChannel::FromInteraction(in);
+
+  // Compute the number of spline knots - use at least 10 knots per decade
+  // && at least 40 knots in the full energy range
+  const double Emin       = 0.01;
```
Comment (mroda88) [inline src/Physics/Resonance/XSection/MKSPPXSecWithCache.cxx:90]: This needs to come from configuration

**Review by mroda88 — COMMENTED** [review]
Things need to be made consistent with configuration requests

File: src/Physics/Resonance/XSection/MKSPPXSecWithCache.cxx (hunk 9 above)
Comment (igor144) [reply src/Physics/Resonance/XSection/MKSPPXSecWithCache.cxx:91]: Marco, I totally forget about it. I changed the default value 40 to 100 for tests. Of course, it has to be changed back to 40.

File: src/Physics/Resonance/XSection/MKSPPXSecWithCache.cxx (hunk 10 above)
Comment (igor144) [reply src/Physics/Resonance/XSection/MKSPPXSecWithCache.cxx:90]: Marco, I just follow the standard approach. The same thing is implemented in _ReinSehgalRESXSecWithCache.cxx_ and _ReinSehgalRESXSecWithCacheFast.cxx_. Then there is a need get it from configuration anywhere.

File: src/Physics/Resonance/XSection/MKSPPXSecWithCache.cxx (hunk 9 above)
Comment (igor144) [reply src/Physics/Resonance/XSection/MKSPPXSecWithCache.cxx:91]: Yes, we can.

**Review by igor144 — APPROVED** [review]

**Review by igor144 — APPROVED** [review]

**Review by igor144 — APPROVED** [review]
All conflicts have been resolved.

**Pull request merged**